example/example.go

```diff
@@ -28,7 +28,7 @@ func main() {
 
 		p := apns.NewPayload()
 		p.APS.Alert.Body = body
-		p.APS.Badge = &badge
+		p.APS.Badge = apns.NewBadgeNumber(badge)
```

Comment:
What about this?

```go
p.APS.Badge.Set(badge)
```

I'm trying to get rid of the `NewBadgeNumber` API. Usually `New*` methods return pointers, and I think we can get away entirely without pointers. I think `Set()` will suffice any place we use `NewBadgeNumber`.